app/models/rule.rb

```diff
@@ -31,6 +31,8 @@ class Rule < ApplicationRecord
     joins(:rule_identifier).where(rule_identifiers: { label: identifier_label })
   }
 
+  default_scope { where(id: RuleResult.pluck(:rule_id)) }
```

Comment:
Notice this chains with the `RuleResult` default scope, so only rules which also contain at least some valid result will be shown. Also notice this is not scoped by host.. which is problematic, as we could:
0 - Assume 2 hosts, host1 and host2 with reports for a profile called `standard`.
1 - `host1.rules` returns *n* rules, according to it's results in the db. 
2 - `host2` report is uploaded, and it contains *n+1* rules because OpenSCAP parsed more rules.
3 - `host2.rules` returns *n+1* rules.
4 - Now `host1.rules` also returns *n+1* rules as we are just looking for rules with valid results, but we don't care which host do they come from. To get all rules for which a host has reports, you need to do `Rule.where(id: RuleResult.where(host: host1).pluck(:rule_id))`. We don't do this on the GraphQL responses.